### ISSUE TITLE
String decoding on pages

### ISSUE THREAD

**Issue by polinasand** [googleinterns/step259-2020#30]
In Datastore we may have entities with string field that contain special characters like `Crêpe` or `блины`. 
On HTML page, these values represent like 
cr�pe
?????
This is not a problem because of datastore, it seems problem occurrs during fetching data.
@grenlayk could you take a look at this, or if you can't I'll fix it after solving a problem with `type` field filling.

**Comment by grenlayk** [comment]
Sure, I will take a look

**Comment by grenlayk** [comment]
@sandatsian  Could you please describe more in which page does this happen?
I checked it on `meal.html` page by adding hard-coded "ingredient" which `encodingCheck()`:
```
const text = "Crêpe блины";
ingredientsElement.appendChild(
    createElementByTag(encodingCheck(capitalizeFirst(text)), "li")
);
```
And it displayed correctly. Probably I missed somewhere this check.

**Comment by grenlayk** [comment]
I tried to solve this problem by searching different libraries on the Net, but didn't found anything.
(you can take a look at [`encodingCheck`](https://github.com/googleinterns/step259-2020/tree/encodingCheck) branch)

@bprusak @zajonc-gugle do you have any ideas?

**Comment by grenlayk** [comment]
If we add something like `Crêpe блины` to database we receive `�bleskiver ?????` by fetch and I do not know how to convert it.

**Comment by zajonc-gugle** [comment]
Debugging Habañero.

The font we use is able to render non-ascii characters.
The `encodingCheck` function doesn't seem to break anything.

One interesting bit is that in the network tab, the response is rendered properly. However, sending a `fetch` request manually returns a messed up string:
```
fetch('meal/39').then(x => x.json()).then(x => console.log(x.description))
//. ... and minced aj� limo ...
```

There's a similar issue reported on SO: https://stackoverflow.com/questions/55282444/javascript-fetch-characters-with-encoding-issues

It seems that our servlet returns its content with the following header:
content-type | application/json;;charset=iso-8859-1

I think setting the content type explicitly in the servlet should be the quickest fix.

**Comment by zajonc-gugle** [comment]
It's also interesting that there's two semicolons in the header. Perhaps we're not supposed to manually add one when setting content-type (we currently set content-type to "application/json;" rather than "application/json"). Maybe this causes some parsing issues and confuses the js decoder?

**Comment by zajonc-gugle** [comment]
Ok, it seems that the trailing semicolon is the root cause. I've pushed by temporary changes that seem to fix the issue here:
https://github.com/googleinterns/step259-2020/tree/zajonc-unicode-fix

The commit includes a few hacks to make the tests pass locally. You can use the commit as a base for a proper fix (i.e. removing the `;` from all content type setters). Let's not use my commit directly, but just create a new one with just the relevant changes.

**Comment by grenlayk** [comment]
Think that this problem is solved by #35
